src/Uno.UI/UI/Xaml/Controls/Panel/Panel.Android.cs

```diff
@@ -20,15 +21,31 @@ namespace Windows.UI.Xaml.Controls
 {
 	public partial class Panel : IEnumerable
 	{
-		private readonly SerialDisposable _backgroundBrushChanged = new SerialDisposable();
-		private readonly SerialDisposable _borderBrushChanged = new SerialDisposable();
+		private WeakBrushChangedProxy _backgroundBrushChangedProxy;
+		private Action _backgroundBrushChanged;
+
+		private WeakBrushChangedProxy _borderBrushChangedProxy;
+		private Action _borderBrushChanged;
 		private BorderLayerRenderer _borderRenderer = new BorderLayerRenderer();
 
 		public Panel()
 		{
 			Initialize();
 		}
 
+		~Panel()
+		{
+			_backgroundBrushChangedProxy?.Unsubscribe();
+			_borderBrushChangedProxy?.Unsubscribe();
+		}
```

Comment:
@Youssef1313 I don't think this is necessary. The proxy itself should be collected and that should be enough for the unsubscription to happen.

Reply:
The brushes hold a reference to the proxy via the event subscription. So long living brushes will cause proxies to leak unless the control calls Unsubscribe.